src/axom/inlet/Inlet.hpp

```diff
@@ -465,9 +454,9 @@ class Inlet : public SchemaCreator
    * \return Whether or not the array was found
    *****************************************************************************
    */
-  bool getBoolArray(std::unordered_map<int, bool>& map)
+  bool getArray(std::unordered_map<int, bool>& map)
```

Comment:
Could the value type by templated? E.g.
```cpp
template<typename T>
bool getArray(std::unordered_map<int, T>& map);
```
Presumably, one would also want an array of user-defined structs.

Reply:
Perhaps that is for a future PR?

Reply:
I had considered (and attempted) this but the `getArray` functions are part of the `SchemaCreator` interface, and template functions cannot be virtual.  I anticipate that adding arrays of user-defined structs (not currently supported) could require some significant changes, so that will probably be in a separate PR.

Reply:
We could remove the SchemaCreator class entirely, I put it there to ease the chaining of nested function calls, but if there is a better way to handle that, I'm perfectly fine with that.  I agree this should be in a separate PR.

Reply:
We could go with how Sidre handles this by not having this logic at the Inlet class level and making the user call something like `auto table = myInlet->getGlobalTable()` then doing the defining.